src/services/authorization.ts

```diff
@@ -26,27 +28,120 @@ const permissionMapping = {
  * @param item
  * @throws if the user cannot access the item
  */
+
+export const validatePermissionMany = async (
+  {
+    itemMembershipRepository,
+    itemTagRepository,
+  }: {
+    itemMembershipRepository: typeof ItemMembershipRepository;
+    itemTagRepository: typeof ItemTagRepository;
+  },
+  permission: PermissionLevel,
+  member: Actor,
+  items: Item[],
+): Promise<ResultOf<ItemMembership | null>> => {
+  // batch request for all items
+  const inheritedMemberships = member
+    ? await itemMembershipRepository.getInheritedMany(items, member, true)
+    : null;
+  const tags = await itemTagRepository.hasManyForMany(items, [
+    ItemTagType.Public,
+    ItemTagType.Hidden,
+  ]);
+
+  const result: ResultOf<ItemMembership | null> = {
+    data: inheritedMemberships?.data ?? {},
+    errors: [],
+  };
+
+  for (const item of items) {
+    const highest = result.data[item.id]?.permission;
+    const isValid = highest && permissionMapping[highest].includes(permission);
+    const isPublic = tags.data[item.id].includes(ItemTagType.Public);
+
+    // HIDDEN CHECK - prevent read
+    // cannot read if your have read access only
+    if (highest === PermissionLevel.Read || (isPublic && !highest)) {
```

Comment:
Q: can members with write access view hidden items?

Reply:
I guess so, so you can change the hidden status? I just kept the existing behavior